[13.0][MIG] l10n_es_aeat_mod111

MiquelRForgeFlow opened the pull request:
Standard migration.

Depende de:

- [x] https://github.com/OCA/l10n-spain/pull/1298.

Commit:
[ADD] l10n_es_aeat_mod111: nuevo módulo para la presentación del modelo 111

Commit:
[IMP] l10n_es_aeat_mod111: completar el módulo para la exportación + ajustar codigo a la nueva API

Commit:
[DEL] l10n_es_aeat_mod111: quitar el módulo del módelo 111 para añadirlo en otro PR

Commit:
[ADD] l10n_es_aeat_mod111: modulo para sacar el modelo 111

Commit:
[IMP] l10n_es_aeat_mod111: metodos nuevos para facilitar los calculos

pedrobaeza committed:
[IMP] l10n_es_aeat_mod111: Refinamientos varios

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Avoid error for author field length

pedrobaeza committed:
[IMP+FIX] l10n_es_aeat_*: Cálculo por códigos de impuestos generalizado + tratamiento residentes 216 y cálculo

pedrobaeza committed:
[FIX] l10n_es_aeat_mod*: Autores/patrocionadores expandidos + PEP8

oca-transbot committed:
OCA Transbot updated translations from Transifex

cubells committed:
[FIX] Corrección cálculo diversos modelos

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Eliminar periodo específico y corregir exportación

pedrobaeza committed:
[IMP] l10n_es_aeat_*: Cálculo de importes de impuestos por cadena en lugar de recordset + gestionar grupos de compañías

antespi committed:
[FIX] l10n_es_aeat_mod111 calcular Total liquidación

antespi committed:
[IMP] l10n_es_aeat_mod111

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Mejor herencia de impuestos

El anterior método utilizado, al tener una clase y un many2one genérico,
no permitía poner los datos en múltiples modelos hijos, ya que se repetían
los IDs, y cogía los datos de un modelo en otros.

Con este nuevo sistema, se sigue utilizando una tabla, pero con doble
índice: el modelo y el ID, parecido a un campo reference, pero gestionado
por código propio en el método de cálculo.

El 303 está adaptado a este nuevo sistema, y se incluye un script de
migración que conserve los datos.

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Evitar problemas en campos related no readonly. Fixes #290

Cuando se tienen campos related, aunque el valor no se cambie,
Odoo lanza una escritura en el objeto relacionado cada vez que se
guarda el objeto principal. En el caso de la moneda asociada a la
compañía, hace que aquellos usuarios que no tengan permiso para
modificar la compañía (la mayoría), falle en la creación del modelo.

oca-transbot committed:
OCA Transbot updated translations from Transifex

jalzaga committed:
Fix Mod. 111 error _get_tax_code_lines

pedrobaeza committed:
[FIX] l10n_es_account_invoice_sequence: PEP8

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[IMP] l10n_es_aeat_mod111: Nuevo formato de exportación del modelo 111 para los ejercicios 2016 y siguientes

cubells committed:
[IMP] Add tax amount to view

Commit:
[IMP] l10n_es_aeat_mod11, calculo rendimientos de trabajo

antespi committed:
l10n_es_aeat_mod111: Migration to 9.0

* Added tests

pedrobaeza committed:
l10n_es_aeat_mod111: Add multi-company security rule

Roodin committed:
[MIG] l10n_es_aeat_mod111: Migrated to 10.0

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Avoid error on multi-company envs

With the limitation of ORM with tax_line_ids in @api.depends,
we avoid the access error this way

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Remove incorrect code

gurneyalex committed:
[REF] i18n: Updating translation terms from weblate [ci skip]

oca-transbot committed:
OCA Transbot updated translations from Transifex

javierjcf committed:
[MIG]l10n_es_aeat_mod111: Migrado a 11

pedrobaeza committed:
[MIG] l10n_es_aeat_mod111: Migration to 12.0

* Standard procedure
* Adapted views

oca-travis committed:
[UPD] Update l10n_es_aeat_mod111.pot

pedrobaeza committed:
[IMP] l10n_es_aeat: Add active field to export configs

This way, we don't initially see all the subconfigs for each model and we can concentrate
on main export configurations. This is got through `active` field default behaviour.
Archived config are still directly searchable through the subconfig selector in the
configuration line.

All models export configs have been adapted to this new approach.

mvrodriguez committed:
Translated using Weblate (Galician)

Currently translated at 13.7% (16 of 117 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat_mod111
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat_mod111/gl/

pedrobaeza committed:
[IMP] l10n_es_aeat: Add chatter + activities to AEAT models

* Include chatter (mail.thread)
* Include activities (mail.activity.mixin)
* Track AEAT report state change
* Include new admin user in AEAT group by default

oca-travis committed:
[UPD] Update l10n_es_aeat_mod111.pot

MiquelRForgeFlow committed:
[IMP] l10n_es_aeat_mod111: black, isort

MiquelRForgeFlow committed:
[MIG] l10n_es_aeat_mod111: Migration to 13.0

HaraldPanten reviewed (changes requested):
Testeado en local con el PR de dependencia l10n_es_aeat.

Tanto con facturas como con asientos manuales, me hace el cálculo correctamente. Sin embargo no consigo que me calcule bien las casillas 1 y 3 número de perceptores.

pedrobaeza commented:
@HaraldPanten has introducido empresa en la columna correspondiente del apunte?

HaraldPanten commented:
Correcto, @pedrobaeza . Adjunto captura del asiento:

<img width="1184" alt="Captura de pantalla" src="https://user-images.githubusercontent.com/32449357/75585072-9375da00-5a71-11ea-9421-1004ca9a7e10.png">

Si hay algo mal, yo ahora mismo no caigo. A vosotros os sale todo OK?

He hecho algunas pruebas más y parece que también falla el cálculo de las casillas 2 y 8. Calcula bien las retenciones, pero no las percepciones.

Actualizado a fecha 02/03 --> Añadir que el fallo se da tanto en asientos manuales como al realizar una "factura".

pedrobaeza commented:
Efectivamente es correcto en principio. Yo no lo he probado aún, así que no te podría decir. Luego prueba el principal y fusionamos, y después vendrá éste.

HaraldPanten commented on l10n_es_aeat_mod111/models/mod111.py:
Hola @MiquelRForgeFlow  ,
hemos encontrado que existe un problema con los cálculos de las casillas al lanzar la acción de 
`def calculate(self)` , debido a un retraso de la escritura de la cache de Odoo en el sistema.
Una manera de arreglarlo sería indicando al principio de cada función de campos calculados el siguiente código:
`self.flush()
 self.invalidate_cache()`

pero obviamente esto no es mantenible... por lo que una sugerencia podría ser cambiar la función base de `def calculate(self)` en el módulo l10n_es_aeat para que realice esas dos funciones y al terminar el cálculo te grabe todo en memoria.
'Archivo: 10n_es_aeat_report.py' Linea: 433

¿Qué pensáis? @pedrobaeza @JordiBForgeFlow @acysos ......

MiquelRForgeFlow replied:
solucionado en el l10n_es_aeat

pedrobaeza committed:
[IMP+FIX] l10n_es_aeat: Move get taxes method to company + cleaning in tests

This way, we can reuse it in other modules that don't inherit AEAT report like SII.

We also clean a non needed reading of the tax description field.

Tests were still using description method for populating data, so we switch to new
system and perform needed changes in dependent tests.

s

pedrobaeza committed:
[IMP] l10n_es_aeat_mod111: README by fragments

Commit:
Create ROADMAP.rst

rlizana committed:
[12.0] l10n_es_aeat_*: [FIX] Caracteres no válidos 'NIF Empresa Desarrollo'

acysos commented:
Funcionalmente es correcto, yo lo he probado.

ValentinVinagre reviewed (approved):
LGTM

pedrobaeza commented on l10n_es_aeat_mod111/readme/CREDITS.rst:
```diff
@@ -0,0 +1,24 @@
+La migración de este módulo forma parte de una campaña de migración de la
```
Esto hay que eliminarlo porque ya para esta versión no hay campaña que valga.

MiquelRForgeFlow replied:
ok, deleted

pedrobaeza replied:
No lo has borrado entero, que es la idea...

pedrobaeza reviewed (changes requested):
Falta el script de migración que renombre `type` a `statement_type`.

pedrobaeza commented:
Sigue faltando el script de migración

MiquelRForgeFlow commented:
@pedrobaeza thanks, ready to merge

HaraldPanten reviewed (approved):
Probado con asiento manual y con facturas y me cuadra. Revisión funcional 👍

pedrobaeza commented:
/ocabot merge nobump
